nucypher/blockchain/eth/networks.py

```diff
@@ -1,57 +1,83 @@
-class NetworksInventory:  # TODO: See #1564
-
-    MAINNET = "mainnet"
-    LYNX = "lynx"
-    IBEX = "ibex"  # this is required for configuration file migrations (backwards compatibility)
-    ETH = "ethereum"
-    TAPIR = "tapir"
-    ORYX = "oryx"
-
-    # TODO: Use naming scheme to preserve multiple compatibility with multiple deployments to a single network?
-    POLYGON = 'polygon'
-    MUMBAI = 'mumbai'
-
-    UNKNOWN = 'unknown'  # TODO: Is there a better way to signal an unknown network?
-    DEFAULT = MAINNET
-
-    __to_chain_id_eth = {
-        MAINNET: 1,  # Ethereum Mainnet
-        IBEX: 5,  # this is required for configuration file migrations (backwards compatibility)
-        LYNX: 5,  # Goerli
-        TAPIR: 5,
-        ORYX: 5
-    }
-    __to_chain_id_polygon = {
-        # TODO: Use naming scheme?
-        POLYGON: 137,    # Polygon Mainnet
-        MUMBAI: 80001,   # Polygon Testnet (Mumbai)
-    }
-
-    ETH_NETWORKS = tuple(__to_chain_id_eth.keys())
-    POLY_NETWORKS = tuple(__to_chain_id_polygon.keys())
-
-    NETWORKS = ETH_NETWORKS + POLY_NETWORKS
-
-    class UnrecognizedNetwork(RuntimeError):
-        pass
+from enum import Enum
+from typing import List, NamedTuple
 
-    @classmethod
-    def get_ethereum_chain_id(cls, network):
-        try:
-            return cls.__to_chain_id_eth[network]
-        except KeyError:
-            raise cls.UnrecognizedNetwork(network)
+from nucypher.config.constants import TEMPORARY_DOMAIN
+
+
+class ChainInfo(NamedTuple):
+    chain_id: int
+    chain_name: str
+
+    # Override eventual Enum name. TODO: better way?
+    @property
+    def name(self) -> str:
+        return self.chain_name
+
+
+class EthNetwork(ChainInfo, Enum):
+    MAINNET = ChainInfo(1, "mainnet")
+    GOERLI = ChainInfo(5, "goerli")
+    SEPOLIA = ChainInfo(11155111, "sepolia")
+    # testing
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class PolyNetwork(ChainInfo, Enum):
+    POLYGON = ChainInfo(137, "polygon")
+    MUMBAI = ChainInfo(80001, "mumbai")
+    # testing
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class TACoNetwork(NamedTuple):
```

Comment:
To help disambiguate between "networks" I suggest using "domain"

```suggestion
class TACoDomain(NamedTuple):
```

Reply:
I renamed [Eth/Poly]Network -> [Eth/Polygon]Chain, so not sure if that changes your mind,.

I think we need to lean in to either "domain" or "network". We've used both across a variety of areas. We've used "domain" for characters / config /  registry (your PR), but "network" for other areas such as: ContractPayment, CLI options, BaseActor, CLI tools.

I am partial to "network" (but this involves updating the config from domain), so I'm game to change it to "domain", but just want to confirm that everything in our codebase related to names like "lynx", "mainnet" etc. will be considered "domain" moving forward.

Reply:
I suggest that we use `domain` as the name for the abstraction representing a set of nucypher contracts deployed across two blockchains and `network` in the industry-conventional manner to indicate a blockchain network (mainnet, sepolia, goerli, polygon-mumbi, etc.).

The `domain` abstraction used in this manner is intentionally the same `domain` used in P2P.  This is sensible since P2P verification of node stakes, bonds, and other on-chain state must be represented in the correct contract set (and thus blockchains) in the `domain`.

Reply:
✔️